toolchain/codegen/codegen.cpp

```diff
@@ -48,24 +43,53 @@ auto PrintAssemblyFromModule(llvm::Module& module,
   std::optional<llvm::Reloc::Model> reloc_model;
   auto* target_machine = target->createTargetMachine(
```

Comment:
Prefer `std::unique_ptr<TargetMachine>` instead of manual memory management with `delete`.